packages/openid4vc/src/openid4vc-verifier/OpenId4VcSiopVerifierService.ts

```diff
@@ -0,0 +1,364 @@
+    private w3cCredentialService: W3cCredentialService,
+    private openId4VcVerifierRepository: OpenId4VcVerifierRepository,
+    private config: OpenId4VcVerifierModuleConfig
+  ) {}
+
+  public async createAuthorizationRequest(
+    agentContext: AgentContext,
+    options: OpenId4VcSiopCreateAuthorizationRequestOptions & { verifier: OpenId4VcVerifierRecord }
+  ): Promise<OpenId4VcSiopCreateAuthorizationRequestReturn> {
+    const nonce = await agentContext.wallet.generateNonce()
+    const state = await agentContext.wallet.generateNonce()
+    const correlationId = utils.uuid()
+
+    const relyingParty = await this.getRelyingParty(agentContext, options.verifier, {
+      presentationDefinition: options.presentationExchange?.definition,
+      requestSigner: options.requestSigner,
+    })
+
+    const authorizationRequest = await relyingParty.createAuthorizationRequest({
+      correlationId,
+      nonce,
+      state,
+    })
+
+    const authorizationRequestUri = await authorizationRequest.uri()
+
+    return {
+      authorizationRequestUri: authorizationRequestUri.encodedUri,
+      authorizationRequestPayload: authorizationRequest.payload,
+    }
+  }
+
+  public async verifyAuthorizationResponse(
+    agentContext: AgentContext,
+    options: OpenId4VcSiopVerifyAuthorizationResponseOptions & { verifier: OpenId4VcVerifierRecord }
+  ): Promise<OpenId4VcSiopVerifiedAuthorizationResponse> {
+    const authorizationResponse = await AuthorizationResponse.fromPayload(options.authorizationResponse).catch(() => {
+      throw new AriesFrameworkError(
+        `Unable to parse authorization response payload. ${JSON.stringify(options.authorizationResponse)}`
+      )
+    })
+
+    const responseNonce = await authorizationResponse.getMergedProperty<string>('nonce', {
+      hasher: Hasher.hash,
+    })
+    const responseState = await authorizationResponse.getMergedProperty<string>('state', {
+      hasher: Hasher.hash,
+    })
+    const sessionManager = this.config.getSessionManager(agentContext)
+
+    const correlationId = responseNonce
+      ? await sessionManager.getCorrelationIdByNonce(responseNonce, false)
+      : responseState
+      ? await sessionManager.getCorrelationIdByState(responseState, false)
+      : undefined
+
+    if (!correlationId) {
+      throw new AriesFrameworkError(
+        `Unable to find correlationId for nonce '${responseNonce}' or state '${responseState}'`
+      )
+    }
+
+    const requestSessionState = await sessionManager.getRequestStateByCorrelationId(correlationId)
+    if (!requestSessionState) {
+      throw new AriesFrameworkError(`Unable to find request state for correlationId '${correlationId}'`)
+    }
+
+    const requestClientId = await requestSessionState.request.getMergedProperty<string>('client_id')
+    const requestNonce = await requestSessionState.request.getMergedProperty<string>('nonce')
+    const requestState = await requestSessionState.request.getMergedProperty<string>('state')
```

Comment:
```suggestion
    const requestClientId = await requestSessionState.request.getMergedProperty<string | undefined>('client_id')
    const requestNonce = await requestSessionState.request.getMergedProperty<string | undefined>('nonce')
    const requestState = await requestSessionState.request.getMergedProperty<string | undefined>('state')
```

Or does the get method return `<T | undefined>`?

Reply:
Yes method returns it as optional type